psq/worker.py

```diff
@@ -42,7 +45,12 @@ def _safe_dequeue(self):
             wait_exponential_multiplier=1000, wait_exponential_max=10000,
             retry_on_exception=lambda e: not isinstance(e, KeyboardInterrupt))
         def inner():
-            return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            try:
+                return self.queue.dequeue(max=self.tasks_per_poll, block=True)
+            except errors.GaxError as e:
```

Comment:
I don't have much detail on what is causing this exception, but I'll take a stab in the dark.

I'd guess you're talking about the DEADLINE_EXCEED on pull? See the ongoing discussion: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2576#issuecomment-262029203